### PR TITLE
Gherkin: Devfile samples should be included in Samples catalog & "searchable" via Topology

### PR DESCRIPTION
**Pull request by sanketpathak**
Epic: https://issues.redhat.com/browse/ODC-22
Jira: https://issues.redhat.com/browse/ODC-5876

Acceptance Criteria:

- As a dev, I should be able to see samples based on devfiles as well as builder images
- As a dev, when using the Quick Add in topology, the search should include samples
- As a dev, when looking at the Dev Catalog, I should be able to see devfiles
- As a dev, when looking at the Dev Catalog, I should see a devfile "type" which brings me to a a devfile subcatalog
- As a dev, I should have a devfile option in the Dev Catalog card on the Add page which brings me directly to the devfile catalog

### CODE REVIEW

**Comment by sanketpathak** [comment]
cc @invincibleJai

**Review by rohitkrai03 — APPROVED** [review]

File: frontend/packages/topology/integration-tests/features/topology/quick-search-topology.feature [hunk 1]
```diff
@@ -82,5 +82,36 @@ Feature: Provide quick search from topology/list views to add to project
...
+
+
+        @odc-22 @to-do
+        Scenario: Quick Add of Devfile Sample in topology graph view: T-02-TC012
+            Given user is at topology graph view
+             When user clicks Add to project button
+              And user enters "node" in Add to project search bar
+              And user selects Basic NodeJS Samples option
+              And user clicks on Create Devfile Samples
+              And user assigns Name as "basic-nodejs-sample-ex1"
```
Comment (makambalaji) [inline frontend/packages/topology/integration-tests/features/topology/quick-search-topology.feature:106]:
```suggestion
              And user enters Name as "basic-nodejs-sample-ex1" in Import from Devfile page
```

File: frontend/packages/topology/integration-tests/features/topology/quick-search-topology.feature [hunk 2]
```diff
@@ -82,5 +82,36 @@ Feature: Provide quick search from topology/list views to add to project
...
+              And user clicks on Create button in Import from Devfile page
+             Then user is taken to the Topology page with workload "basic-nodejs-sample-ex1" created
+
+
+        @to-do
```
Comment (makambalaji) [inline frontend/packages/topology/integration-tests/features/topology/quick-search-topology.feature:111]: Please add @odc-22, so we get to know. this is related to that particular epic

**Review by makambalaji — CHANGES_REQUESTED** [review]

File: frontend/packages/topology/integration-tests/features/topology/quick-search-topology.feature [hunk 3]
```diff
@@ -55,7 +55,7 @@ Feature: Provide quick search from topology/list views to add to project
             Given user is at topology graph view
              When user clicks Add to project button
               And user enters "django" in Add to project search bar
-              And user clicks on View all results option
+              And user clicks on View all developer catalog items option
```
[the remark this answers is not in the document]
Comment (sanketpathak) [reply frontend/packages/topology/integration-tests/features/topology/quick-search-topology.feature:58]: This is not related to odc-22, I've updated it as per the ux

File: frontend/packages/topology/integration-tests/features/topology/quick-search-topology.feature [hunk 4]
```diff
@@ -73,7 +73,7 @@ Feature: Provide quick search from topology/list views to add to project
               And user enters "monitoring" in Add to project search bar
```
[the remark this answers is not in the document]
Comment (sanketpathak) [reply frontend/packages/topology/integration-tests/features/topology/quick-search-topology.feature:73]: This is not part of odc-22

File: frontend/packages/dev-console/integration-tests/features/addFlow/developer-catalog-details.feature [hunk 5]
```diff
@@ -15,7 +15,7 @@ Feature: Developer Catalog Page
              When user clicks on From Catalog card
```
[the remark this answers is not in the document]
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/addFlow/developer-catalog-details.feature:15]: Yes, I've updated it

File: frontend/packages/dev-console/integration-tests/features/addFlow/developer-catalog-details.feature [hunk 6]
```diff
@@ -15,7 +15,7 @@ Feature: Developer Catalog Page
...
              Then user will see All Items already selected
               And user will see CICD, Databases, Languages, Middleware, Other categories
-              And user will see Builder Images, Event Sources, Helm Charts, Operator Backed, Templates types
+              And user will see Builder Images, Devfiles,Event Sources, Helm Charts, Operator Backed, Templates types
```
[the remark this answers is not in the document]
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/addFlow/developer-catalog-details.feature:18]: Updated

File: frontend/packages/topology/integration-tests/features/topology/quick-search-topology.feature [hunk 7]
```diff
@@ -73,7 +73,7 @@ Feature: Provide quick search from topology/list views to add to project
...
               And user selects Monitoring your sample application option
```
[the remark this answers is not in the document]
Comment (sanketpathak) [reply frontend/packages/topology/integration-tests/features/topology/quick-search-topology.feature:74]: Updated

File: frontend/packages/topology/integration-tests/features/topology/quick-search-topology.feature [hunk 8]
```diff
@@ -73,7 +73,7 @@ Feature: Provide quick search from topology/list views to add to project
...
               And user clicks on Start
-             Then Monitoring your sample application quick start starts in context of the Topology.
+             Then Monitoring your sample application quick start starts in context of the Topology
```
[the remark this answers is not in the document]
Comment (sanketpathak) [reply frontend/packages/topology/integration-tests/features/topology/quick-search-topology.feature:76]: Updated

File: frontend/packages/topology/integration-tests/features/topology/quick-search-topology.feature [hunk 9]
```diff
@@ -82,5 +82,36 @@ Feature: Provide quick search from topology/list views to add to project
              When user clicks Add to project button
               And user enters "monitoring" in Add to project search bar
```
[the remark this answers is not in the document]
Comment (sanketpathak) [reply frontend/packages/topology/integration-tests/features/topology/quick-search-topology.feature:83]: Updated with a new entry

File: frontend/packages/topology/integration-tests/features/topology/quick-search-topology.feature [hunk 10]
```diff
@@ -82,5 +82,36 @@ Feature: Provide quick search from topology/list views to add to project
...
               And user clicks on View all Quick Starts option
-             Then user is taken to the search results in context of the Quick Starts Catalog.
+             Then user is taken to the search results in context of the Quick Starts Catalog
```
[the remark this answers is not in the document]
Comment (sanketpathak) [reply frontend/packages/topology/integration-tests/features/topology/quick-search-topology.feature:85]: Updated

File: frontend/packages/topology/integration-tests/features/topology/quick-search-topology.feature (hunk 1 above)
Comment (sanketpathak) [reply frontend/packages/topology/integration-tests/features/topology/quick-search-topology.feature:106]: Updated

File: frontend/packages/topology/integration-tests/features/topology/quick-search-topology.feature (hunk 2 above)
Comment (sanketpathak) [reply frontend/packages/topology/integration-tests/features/topology/quick-search-topology.feature:111]: Added

**Pull request merged**